components/GameScreen/GeneratedMusic.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react"
+import { Contract, ethers } from "ethers"
+import Image from "next/image"
+import abi from "../../lib/abi-musicGame.json"
+import getIpfsLink from "../../lib/getIpfsLink"
+
+const GeneratedMusic = ({ checked, choices, onClickHandler }: any) => {
+  const [trackMetadata, setTrackMetadata] = useState([])
+
+  const getProvider = () => {
+    const goerliRpc = "https://ethereum-goerli-rpc.allthatnode.com"
+    const provider = ethers.getDefaultProvider(goerliRpc)
+    return provider
+  }
+
+  useEffect(() => {
+    const init = async () => {
+      const contract = new Contract(
+        process.env.NEXT_PUBLIC_MUSIC_GAME_CONTRACT_ADDRESS,
+        abi,
+        getProvider(),
+      )
+      const tokens = await contract.cre8ingTokens()
+      const uris = await contract.cre8ingURI()
+      const tokensFiltered = tokens.filter((token) => token.toString() !== "0")
+      const urisFiltered = uris.filter((uri) => uri !== "")
+      const urisDecoded = urisFiltered.map((uri, index) => {
+        const sub = uri.substring(uri.indexOf(",") + 1)
+        return { ...JSON.parse(window.atob(sub)), tokenId: tokensFiltered[index] }
+      })
+      setTrackMetadata(urisDecoded)
+    }
+
```

Comment:
should probably move this entire process to an api endpoint

Reply:
moved to API endpoint `/api/getStakedTracks` ✅ 

![Screenshot 2023-02-28 at 4 54 29 PM](https://user-images.githubusercontent.com/23249402/221964140-b31fbeec-7324-4fff-9f1a-0b608ba8af38.png)